code/game/machinery/OpTable.dm

```diff
@@ -71,13 +88,18 @@
 	return
 
 /obj/machinery/optable/proc/check_victim()
-	if(locate(/mob/living/carbon/human, src.loc))
-		var/mob/living/carbon/human/M = locate(/mob/living/carbon/human, src.loc)
-		if(M.lying)
-			src.victim = M
-			icon_state = M.pulse ? "[modify_state]-active" : "[modify_state]-idle"
-			return 1
-	src.victim = null
+	if(!victim || !victim.lying || victim.loc != loc)
+		suppressing = FALSE
+		victim = null
+		if(locate(/mob/living/carbon/human) in loc)

```

Comment:
hnrg.

You either use locate once, or you iterate over all. You don't need to do both.
```dm
var/mob/living/carbon/human/H = locate() in loc
if (istype(H))
	if (H.lying)
		icon_state = //bladi bla
		victim = H
```

Reply:
addressed